### PR TITLE
reduce 3.0 vs 4.0 confusion

### PR DESCRIPTION
**Pull request by christophs78**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -137,7 +139,7 @@ Optionally, register the `FacesServlet` in a Servlet 5.0+ compatible deployment
...
 
-The Faces deployment descriptor file `/WEB-INF/faces-config.xml` is fully optional, but if any it must be Jakarta Faces 3.0 compatible, otherwise Jakarta Faces 3.0 will run in a fallback modus matching the exact `version` as declared in `<faces-config>` root element.
+The Faces deployment descriptor file `/WEB-INF/faces-config.xml` is fully optional, but if any it must be Jakarta Faces 3.0+ compatible, otherwise Jakarta Faces 4.0 will run in a fallback modus matching the exact `version` as declared in `<faces-config>` root element.
```
Comment (christophs78) [inline README.md:142]: Is this correct?

File: README.md [hunk 2]
```diff
@@ -220,9 +222,9 @@ Finally create a [Facelets][20] file `/hello.xhtml` as below:
...
+## Activating CDI in Jakarta Faces 4.0
 
-By default, Jakarta Faces 3.0 will run in Jakarta Faces 2.2 modus as to CDI support. Even when you use a Jakarta Faces 2.3 or 3.0 compatible `faces-config.xml`. In other words, the Jakarta Faces 2.3+ feature of injection and EL resolving of Jakarta Faces artifacts ([spec issue 1316](https://github.com/javaee/javaserverfaces-spec/issues/1316)) won't work until you explicitly activate this. In other words, `@Inject FacesContext` doesn't work by default. This was necessary in order for Jakarta Faces 2.3 to stay compatible with 2.2, and this carried over to 3.0.
+By default, Jakarta Faces 4.0 will run in Jakarta Faces 2.2 modus as to CDI support. Even when you use a Jakarta Faces 2.3 or 3.0 compatible `faces-config.xml`. In other words, the Jakarta Faces 2.3+ feature of injection and EL resolving of Jakarta Faces artifacts ([spec issue 1316](https://github.com/javaee/javaserverfaces-spec/issues/1316)) won't work until you explicitly activate this. In other words, `@Inject FacesContext` doesn't work by default. This was necessary in order for Jakarta Faces 2.3 to stay compatible with 2.2, and this carried over to 3.0.
```
Comment (christophs78) [inline README.md:227]: Is this still right for JSF 4.0?

Comment (arjantijms) [reply README.md:227]: Nope, Faces 4.0 will definitely not have a fake Faces 2.2 mode ;) It wil default to just Faces 4.0.

Comment (christophs78) [reply README.md]: Thanks for your reply. Just updated this. Feel free to further modify this description. :-)

Day-job-question: Some quick answer how this works together with https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/jsf/el/SpringBeanFacesELResolver.html?
(We already added Weld a few years ago to fulfil CDI-requirement for Omnifaces.)

**Comment by BalusC** [comment]
Awesome.

File: README.md [hunk 3]
```diff
@@ -16,8 +16,10 @@
 
 # Mojarra 4.0 WIP
 
-Eclipse's implementation of the upcoming Jakarta Faces 4.0 specification 
+Eclipse's implementation of the upcoming Jakarta Faces 4.0 specification
 
+For Mojarra / JSF 2.3 please have a look at https://github.com/eclipse-ee4j/mojarra/blob/2.3/README.md. <br/>
```
Comment (BalusC) [inline README.md:21]: HTML in markdown is itchy, use 2 spaces instead of <br> ;)

File: README.md (hunk 1 above)
Comment (BalusC) [reply README.md:142]: Nope, the `3.0+` part should also say `4.0`.

File: README.md [hunk 4]
```diff
@@ -220,19 +222,9 @@ Finally create a [Facelets][20] file `/hello.xhtml` as below:
...
-There is currently only one way to activate CDI in Jakarta Faces 2.3 and 3.0 and herewith make Jakarta Faces 2.3 and 3.0 to run in full Jakarta Faces 2.3/3.0 modus. Put the `@FacesConfig` annotation on an arbitrary CDI managed bean. For example, a general startup/configuration bean.
-
-```
-@FacesConfig
-@ApplicationScoped
-public class YourApplicationConfig {
-    // ...
-}
-```
+CDI is activated by default in Jakarta Faces 4.0 and can´t be deactivated. It´s not required anymore to add `@FacesConfig` to a CDI managed bean.
```
Comment (BalusC) [inline README.md:227]: I would add that using `@FacesConfig` only removes the need to add a `FacesServlet` entry to `web.xml`.

**Pull request merged**